dali/imgcodec/parsers/webp.cc

```diff
@@ -12,19 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include "dali/imgcodec/parsers/webp.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+  bool is_pattern_matching(const uint8_t *data, const std::string &pattern) {
+    for (size_t i = 0; i < pattern.size(); i++)
+      if (data[i] != pattern[i])
+        return false;
+    return true;
+  }
+
+  bool is_valid_sync_code_lossy(const uint8_t *data) {
+    return data[0] == 0x9D && data[1] == 0x01 && data[2] == 0x2A;
+  }
+
+  const uint8_t SIGNATURE_BYTE_LOSTLESS = 0x2F;
+}  // namespace
+
 ImageInfo WebpParser::GetInfo(ImageSource *encoded) const {
+  auto stream = encoded->Open();
+  ssize_t length = stream->Size();
+
+  // Skipping 12 bytes of the RIFF header chunk
+  // "RIFF" (4 bytes)
+  // File Size (4 bytes)
+  // "WEBP" (4 bytes)
+  stream->SeekRead(12, SEEK_CUR);
+
   ImageInfo info;
-  // info.shape = {h, w, c};   // TODO(janton)
+  const auto header = stream->ReadOne<std::array<uint8_t, 4>>();
+  if (is_pattern_matching(header.data(), "VP8 ")) {
+    // Simple file format (lossy)
+    // https://datatracker.ietf.org/doc/html/rfc6386#section-9.1
+    // Skipping chunk size (4 bytes) and frame tag (3 bytes)
+    stream->SeekRead(7, SEEK_CUR);
+
+    // Verify sync code
+    const auto sync_code = stream->ReadOne<std::array<uint8_t, 3>>();
+    if (!is_valid_sync_code_lossy(sync_code.data())) {
+      DALI_FAIL("Sync code 157 1 42 not found at expected position. Found " +
+                std::to_string(sync_code[0]) + " " +
+                std::to_string(sync_code[1]) + " " +
+                std::to_string(sync_code[2]) + " instead.");
```

Comment:
```suggestion
      DALI_FAIL(make_string("Sync code 157 1 42 not found at expected position. Found ",
                                             sync_code[0], " ", sync_code[1], " ",  sync_code[2], " instead."));
```
Is usually preferred.